src/perf/bin/appmain.cpp

```diff
@@ -0,0 +1,204 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    QUIC Perf Main execution runner.
+
+--*/
+
+#ifdef QUIC_CLOG
+#include "appmain.cpp.clog.h"
+#endif
+
+#define QUIC_TEST_APIS 1
+#include "quic_driver_main.h"
+#include "PerfHelpers.h"
+#include <quic_trace.h>
+
+#ifdef _WIN32
+
+//
+// Name of the driver service for quicperf.sys.
+// Must be defined before quic_driver_helpers.h is included
+//
+#define QUIC_DRIVER_NAME   "quicperf"
+
+#include <winioctl.h>
+#include "perfioctls.h"
+#include "quic_driver_helpers.h"
+
+#endif
+
+extern "C" _IRQL_requires_max_(PASSIVE_LEVEL) void QuicTraceRundown(void) { }
+
+int
+QuicUserMain(
+    _In_ int argc,
+    _In_reads_(argc) _Null_terminated_ char* argv[],
+    _In_ bool KeyboardWait
+    ) {
+    QUIC_EVENT StopEvent;
+    QuicEventInitialize(&StopEvent, true, false);
+
+    int RetVal = QuicMainStart(argc, argv, StopEvent);
+    if (RetVal != 0) {
+        return RetVal;
+    }
+    printf("Ready For Connections!\n\n");
+    fflush(stdout);
+    if (KeyboardWait) {
+        printf("Press enter to exit\n");
+        getchar();
+        QuicEventSet(StopEvent);
+    }
+    RetVal = QuicMainStop(0);
+    QuicEventUninitialize(StopEvent);
+    return RetVal;
+}
+
+#ifdef _WIN32
+
+int
+QuicKernelMain(
+    _In_ int argc,
+    _In_reads_(argc) _Null_terminated_ char* argv[],
+    _In_ bool KeyboardWait,
+    _In_ QUIC_SEC_CONFIG_PARAMS* SelfSignedParams
+    ) {
+    size_t TotalLength = 0;
+
+    // Get total length
+    for (int i = 0; i < argc; ++i) {
+        //
+        // Length of string, plus null terminator, plus length
```

Comment:
Why do you need a length? Generally a double null terminator indicates the end of a list like this:
```
-kernel\0-kbwait\0\0
```

Reply:
I'm going to need to allocate an array on the other side, and having an explicit length means I can do it in 1 loop rather then 2. Also I've never seen double null indicating the end up a list.

Reply:
Fixed.